### PR TITLE
Add optional debug logging and refactors

### PR DESCRIPTION
**Pull request by Fenthick**
Refactors extracted from incoming implementation of Tests resource
There are some functional changes, such as optional debug logging

Issue: KNTK-245

### CODE REVIEW

**Review by mmac-m3a — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -6,3 +6,5 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.6.1
 	github.com/kentik/community_sdk_golang/apiv6 v0.0.0-20210503202321-1dab307d87ff
 )
+
+replace github.com/kentik/community_sdk_golang/apiv6 => github.com/Fenthick/community_sdk_golang/apiv6 v0.0.0-20210531094034-d65e4592bcb6
```
Comment (mateuszmidor) [inline go.mod:10]: If this is related to stub server update (that has been mergerd already), I think it can be removed

**Review by mateuszmidor — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (Fenthick) [reply go.mod:10]: I'm on it.

Comment (Fenthick) [reply go.mod:10]: Actually, already done in other PR.

**Pull request merged**